src/transactions/ChangeTrustTests.cpp

```diff
@@ -131,7 +129,7 @@ TEST_CASE("change trust", "[tx][changetrust]")
             validateTrustLineIsConst();
         });
 
-        for_versions_from(3, app, [&] {
+        for_versions_from(3, *app, [&] {
```

Comment:
wouldn't it be simpler to change what for_versions_*** accepts from reference to pointer?

Reply:
This was an approach I considered, but since I was already changing `app.` to `app->` and `app` to `*app` in so many places I thought I would just do it everywhere. If you and @MonsieurNicolas prefer I change the interface of `for_versions` then I am happy to do so. Another approach would be to just write code like
```
auto appPtr = createTestApplication(...);
auto& app = *app;
```
I like this solution less, but it would result in a smaller change. Let me know what you both prefer.

Reply:
yeah I'm fine with the way it's done right now: passing a pointer implies that you can pass null for some reason which is not the case.

Reply:
I'm fine with both ways, I was just wondering if you really like adding hundreds of '*' into code ;)